packages/aws-cdk-lib/pipelines/test/codepipeline/codepipeline.test.ts

```diff
@@ -197,6 +198,28 @@ test('CodeBuild action role has the right AssumeRolePolicyDocument', () => {
   });
 });
 
+test('CodeBuild asset role has the right Principal', () => {
```

Comment:
nit: can we have another test to strictly test when the flag is false and the role has the root account principal?

Reply:
yep added another test to cover that